### PR TITLE
Add SDPA pattern for HuggingFace models BF16

### PR DESCRIPTION
**Pull request by Valentine233**
### Description

- Add pattern for bf16 input type with fp32 attention mask. (Example model: ElectraForCausalLM)
- Add pattern with batch_size=1 to avoid some clones in graph. (Example model: text-classification+prajjwal1-bert-tiny)

### Newly matched models
Dtype: bf16, machine: SPR

#### Dynamo HuggingFace models

- ElectraForCausalLM (speedup=2.09x)
- ElectraForQuestionAnswering (speedup=4.22x)
- AlbertForQuestionAnswering (speedup=1.36x)
- AlbertForMaskedLM (speedup=1.39x)

#### OOB HuggingFace models

- multiple-choice+google-electra-base-discriminator
- text-classification+prajjwal1-bert-tiny
- text-classification+prajjwal1-bert-mini
- text-classification+google-electra-base-generator
- text-classification+bert-large-cased
- casual-language-modeling+xlm-roberta-base
- text-classification+roberta-base
- text-classification+xlm-roberta-base
- text-classification+albert-base-v2
- token-classification+google-electra-base-generator
- masked-language-modeling+bert-base-cased

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/fx_passes/fuse_attention.py [hunk 1]
```diff
@@ -546,16 +555,26 @@ def _get_sfdp_patterns():
         torch.empty, (1024, 128, 128), device=device, requires_grad=True
     )
 
+    # when batch_size = 1, some clones disappear due to no memory layout change
```
Comment (leslie-fang-intel) [inline torch/_inductor/fx_passes/fuse_attention.py:558]: Can we describe the graph difference (w/wo clone) between bs1 and bsN in more details?

File: torch/_inductor/fx_passes/serialized_patterns/_sfdp_pattern_1.py [hunk 2]
```diff
@@ -179,4 +179,4 @@
 expand_default_3 = CallFunction(aten.expand.default, KeywordArg('value'), Ignored())
 view_default_4 = CallFunction(aten.view.default, expand_default_3, Ignored())
 bmm_default_1 = CallFunction(aten.bmm.default, view_default_3, view_default_4)
-_sfdp_pattern_1_inference_half = CallFunction(aten.view.default, bmm_default_1, Ignored())
+_sfdp_pattern_1_half_inference = CallFunction(aten.view.default, bmm_default_1, Ignored())
```
Comment (sanchitintel) [inline torch/_inductor/fx_passes/serialized_patterns/_sfdp_pattern_1.py:182]: nit: why are you changing names?

Comment (sanchitintel) [reply torch/_inductor/fx_passes/serialized_patterns/_sfdp_pattern_1.py:182]: Oh, I see now

File: torch/_inductor/fx_passes/fuse_attention.py (hunk 1 above)
Comment (Valentine233) [reply torch/_inductor/fx_passes/fuse_attention.py]: Thanks and modified！

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]
cc @eellison
